### PR TITLE
Support BIGINT Numeric Type

### PR DESCRIPTION
**Pull request by dey4ss**
This PR adds `BIGINT` as an alias for `LONG`. `BIGINT` column types in `CREATE TABLE` statements or `CAST`statements are translated to `DataType::Long`, as the two types are equal according to Postgres [1] (8 byte integer). 

[1] https://www.postgresql.org/docs/14/datatype-numeric.html

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -47,12 +47,13 @@
...
-[submodule "third_party/sql-parser"]
-	path = third_party/sql-parser
-	url = https://github.com/hyrise/sql-parser.git
 [submodule "third_party/tpcds-kit"]
 	path = third_party/tpcds-kit
 	url = https://github.com/hyrise-mp/tpcds-kit.git
 [submodule "third_party/zstd"]
 	path = third_party/zstd
 	url = https://github.com/facebook/zstd.git
+[submodule "third_party/sql-parser"]
```
Comment (Bouncner) [inline .gitmodules:56]: Why the change? I think the module have been sorted alphabetically before.

Comment (dey4ss) [reply .gitmodules:56]: I only changed the submodules using GIT CLI to include the parser's feature branch with the `BIGINT` type, the submodules will be reverted once this feature branch is merged into the parser's master.

File: src/test/lib/sql/sql_translator_test.cpp [hunk 2]
```diff
@@ -3125,6 +3127,17 @@ TEST_F(SQLTranslatorTest, CastStatement) {
...
     EXPECT_LQP_EQ(actual_lqp, expected_lqp);
   }
+  {
+    // BIGINT is an alias for LONG
+    const auto cast_expression = expression_vector(cast_(value_(int32_t{1}), DataType::Long));
+    // clang-format off
+    const auto expected_lqp =
+      ProjectionNode::make(cast_expression,
+        DummyTableNode::make());
+    // clang-format on
```
Comment (julianmenzler) [inline src/test/lib/sql/sql_translator_test.cpp:3137]:
```suggestion
    const auto expected_lqp =
    ProjectionNode::make(cast_expression,
      DummyTableNode::make());
    // clang-format on
```

**Review by julianmenzler — APPROVED** [review]

File: src/test/lib/sql/sql_translator_test.cpp (hunk 2 above)
Comment (dey4ss) [reply src/test/lib/sql/sql_translator_test.cpp:3137]: Hm, we are inconsistently regarding the spacing in this file. 
`\s{2}const auto expected_lqp.+\n[ ]{4}` --> 30 matches
`\s{2}const auto expected_lqp.+\n[ ]{2}` --> 143 matches

However, as mostly the unindented version is used, I'm going to change all occurrences according to your suggested change.

**Review by julianmenzler — APPROVED** [review]
Great!
